File: avalon-modules/avalon-game/avalon-game-calculator/src/main/java/io/github/lishangbu/avalon/game/calculator/growthrate/MediumGrowthRateCalculator.java
package io.github.lishangbu.avalon.game.calculator.growthrate;

import org.springframework.stereotype.Service;

/// 较快组成长速率计算器
///
/// @author lishangbu
/// @since 2026/2/25
@Service
public class MediumGrowthRateCalculator extends AbstractGrowthRateCalculator {
    /// 获取较快组成长速率的内部名称
    @Override
    protected String getGrowthRateInternalName() {
        return "medium";
    }

    /// 较快组：
    /// EXP = Lv^3
    @Override
    protected int tryCalculateGrowthRate(int level) {
        return level * level * level;
    }
}

File: avalon-modules/avalon-game/avalon-game-calculator/src/main/kotlin/io/github/lishangbu/avalon/game/calculator/growthrate/MediumGrowthRateCalculator.kt
package io.github.lishangbu.avalon.game.calculator.growthrate

import org.springframework.stereotype.Service

/**
 * 较快组成长速率计算器
 *
 * @author lishangbu
 * @since 2026/2/25
 */
@Service
class MediumGrowthRateCalculator : AbstractGrowthRateCalculator() {
    /** 获取成长速率内部名称 */
    override fun getGrowthRateInternalName(): String = "medium"

    /** 计算较快组在指定等级下的经验值 */
    override fun tryCalculateGrowthRate(level: Int): Int = level * level * level
}
